### DISCUSSION TITLE
It'd be really awesome to be able to generate that code from the db scheme

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4307] [Feedback]
A user says:
>I think it's a really great idea to quickly spin up an application. 
>
>However, I have and probably most always be adverse to using a black boxed framework for anything complex though since it hides away any ability to customise how something works. Even if it's to my own detriment, it's nice to have the option. For the most part though I suppose it does provide the means to come up with a creative solution. By no means have I met this complexity with Dark yet, but I'm not exactly building enterprise software with it... perhaps yet :)
>
>I do have some feedback and possibly a bug:
>
>Bug first:
>
>So here's a simple post event. My problem came in when trying to parse the Date sent in by data... It took a bit of clicking around until it accepted it for some reason. Then while actually in the dbset function it wouldn allow me to type "Date::parsev2" but then it wouldn't pick up my data variable, and even if I typed it in, when I clicked out it would clear the parameter. Please see the gif below to illustrate the point. I haven't encountered this with anything else yet.
>
>Enhancement:
>
>It's more really a nice to have. To take the complexity out of something like the above, instead of typing out all those fields. It'd be really awesome to be able to generate that code from the db scheme.
>
>Like, given a db scheme, in the hamburger menu, having the ability to "generate" really simplistic get, post, delete handlers. These would then just do something like the above, since it's almost a replica of my schema.
>
>I hope this helps, even if just a little.

**Comment by StachuDotNet** [comment]
> However, I have and probably most always be adverse to using a black boxed framework for anything complex though since it hides away any ability to customise how something works. Even if it's to my own detriment, it's nice to have the option. 

I'd like to hear more of what was meant here. I'm not sure what kind of 'black box' there is.
At this point, Dark's source code is available to read and contribute to, which may have not been the case at the time of the feedback?

**Comment by StachuDotNet** [comment]
> Enhancement:
> It's more really a nice to have. To take the complexity out of something like the above, instead of typing out all those fields. It'd be really awesome to be able to generate that code from the db scheme.
> Like, given a db scheme, in the hamburger menu, having the ability to "generate" really simplistic get, post, delete handlers. These would then just do something like the above, since it's almost a replica of my schema.

I like this.
I'm not sure if it's something we'd want to support in Dark itself, but perhaps some day such functionality might be able to be provided by a Package - what do you think, @pbiggar?

**Reply by pbiggar** [reply]
Definitely something we'd want built-in imo.
